src/EditorFeatures/Core/EditAndContinue/PdbMatchingSourceTextProvider.cs

```diff
@@ -30,14 +30,20 @@ internal sealed class PdbMatchingSourceTextProvider() : IEventListener, IPdbMatc
     private bool _isActive;
     private int _baselineSolutionVersion;
     private readonly Dictionary<string, (DocumentState state, int solutionVersion)> _documentsWithChangedLoaderByPath = [];
+    private WorkspaceEventRegistration? _workspaceChangedDisposer;
 
     public void StartListening(Workspace workspace)
-        => workspace.WorkspaceChanged += WorkspaceChanged;
+    {
+        _workspaceChangedDisposer = workspace.RegisterWorkspaceChangedHandler(WorkspaceChanged);
```

Comment:
Maybe add a Debug.Assert if the disposer is non-null? Before a double start/stop could kinda work, but now we'll leak the disposer for good.

Reply:
sounds good